### PR TITLE
[Infra] add configuration for VSSDK thread analyzers

### LINKED ISSUE #4900: [Infra] Adding configuration for VSSDK threading analyzers to remove FPs
### Description

We use our own testable wrappers around the VS thread handling classes. This causes the VSSDK threading analyzers to incorrectly report issues against our code as they are unaware of our equivalent thread-handling methods.

The VS SDK analyzers are customisable - see https://github.com/microsoft/vs-threading/blob/main/doc/analyzers/configuration.md.

We already have customisation for "methods that assert the UI thread". We can add configuration for more methods to reduce the number of false positives further.

TODO
 - [ ] check which analyzers can be configured
 - [ ] add suitable configuration files

### PR DESCRIPTION
**Pull request by duncanp-sonar**
Fixes #4900

### CODE REVIEW

File: SonarLint.VisualStudio.Integration.sln [hunk 1]
```diff
@@ -162,6 +162,12 @@ Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "SonarQube.Client", "src\Son
 EndProject
 Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "SonarQube.Client.Tests", "src\SonarQube.Client.Tests\SonarQube.Client.Tests.csproj", "{4F5AE2A8-56DD-44EC-9872-B7884CC1C1DB}"
 EndProject
+Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "VsAnalyzerConfig", "VsAnalyzerConfig", "{682C440B-E506-4811-9569-9734D8DF4620}"
+	ProjectSection(SolutionItems) = preProject
+		src\vs-threading.MainThreadAssertingMethods.txt = src\vs-threading.MainThreadAssertingMethods.txt
+		src\vs-threading.MainThreadSwitchingMethods.txt = src\vs-threading.MainThreadSwitchingMethods.txt
+	EndProjectSection
+EndProject
```
Comment (duncanp-sonar) [inline SonarLint.VisualStudio.Integration.sln:170]: Added the VSSDK analyzer config files to the solution to make them easier to find.

File: src/Directory.Build.props [hunk 2]
```diff
@@ -13,8 +13,11 @@
     <Compile Include="$(MSBuildThisFileDirectory)AssemblyInfo.Shared.cs">
       <Link>Properties\AssemblyInfo.Shared.cs</Link>
     </Compile>
-	<AdditionalFiles Include="$(MSBuildThisFileDirectory)vs-threading.MainThreadAssertingMethods.txt">
-	      <Link>Properties\vs-threading.MainThreadAssertingMethods.txt</Link>
+    <AdditionalFiles Include="$(MSBuildThisFileDirectory)vs-threading.MainThreadAssertingMethods.txt">
+	    <Visible>false</Visible>
```
Comment (duncanp-sonar) [inline src/Directory.Build.props:17]: Miscellaneous: hide the file in the Solution Explorer so it doesn't appear in every project.

File: src/Directory.Build.props [hunk 3]
```diff
@@ -13,8 +13,11 @@
     <Compile Include="$(MSBuildThisFileDirectory)AssemblyInfo.Shared.cs">
       <Link>Properties\AssemblyInfo.Shared.cs</Link>
     </Compile>
-	<AdditionalFiles Include="$(MSBuildThisFileDirectory)vs-threading.MainThreadAssertingMethods.txt">
-	      <Link>Properties\vs-threading.MainThreadAssertingMethods.txt</Link>
```
Comment (duncanp-sonar) [inline src/Directory.Build.props:17]: Miscellaneous: the `Link` wasn't required - it was telling VS look in a different place for the file.

File: src/Directory.Build.props [hunk 4]
```diff
@@ -13,8 +13,11 @@
...
+    <AdditionalFiles Include="$(MSBuildThisFileDirectory)vs-threading.MainThreadAssertingMethods.txt">
+	    <Visible>false</Visible>
+    </AdditionalFiles>
+    <AdditionalFiles Include="$(MSBuildThisFileDirectory)vs-threading.MainThreadSwitchingMethods.txt">
+      <Visible>false</Visible>
```
Comment (duncanp-sonar) [inline src/Directory.Build.props:20]: Add a reference to the new config file.

File: src/vs-threading.MainThreadAssertingMethods.txt [hunk 5]
```diff
@@ -1 +1,4 @@
-[SonarLint.VisualStudio.Core.IThreadHandling]::ThrowIfNotOnUIThread
\ No newline at end of file
+# List all of our methods that assert we are on the UI thread.
+# Format: see https://github.com/microsoft/vs-threading/blob/main/doc/analyzers/configuration.md#methods-that-assert-the-main-thread
```
Comment (duncanp-sonar) [inline src/vs-threading.MainThreadAssertingMethods.txt:2]: Miscellaneous: added some comments to the existing config file to help future maintainers.

File: src/vs-threading.MainThreadSwitchingMethods.txt [hunk 6]
```diff
@@ -0,0 +1,7 @@
+# List all of our methods that handle switching the UI thread.
+# Format: see https://github.com/microsoft/vs-threading/blob/main/doc/analyzers/configuration.md#methods-that-switch-to-the-main-thread
+
+[SonarLint.VisualStudio.Core.IThreadHandling]::SwitchToMainThreadAsync
+[SonarLint.VisualStudio.Core.IThreadHandling]::RunOnUIThread
+[SonarLint.VisualStudio.Core.IThreadHandling]::RunOnUIThreadAsync
+
```
Comment (duncanp-sonar) [inline src/vs-threading.MainThreadSwitchingMethods.txt:7]: New config file. Unfortunately, it's only reduce the number of issues by three or four.

**Review by georgii-borovinskikh-sonarsource — APPROVED** [review]

**Pull request merged**